feg/cloud/go/services/feg_relay/servicers/feg_to_gw_relay.go

```diff
@@ -53,37 +53,36 @@ func getHwIDFromIMSI(ctx context.Context, imsi string) (string, error) {
 		return "", err
 	}
 	for _, nid := range servedIds {
-		hwId, err := directoryd.GetHWIDForIMSI(nid, imsi)
-		if err == nil && len(hwId) != 0 {
+		hwID, err := directoryd.GetHWIDForIMSI(nid, imsi)
+		if err == nil && len(hwID) != 0 {
 			glog.V(2).Infof("IMSI to send is %v\n", imsi)
-			return hwId, nil
+			return hwID, nil
 		}
 	}
 	return "", fmt.Errorf("could not find gateway location for IMSI: %s", imsi)
 }
 
-func validateFegContext(ctx context.Context) error {
-	fegId := protos.GetClientGateway(ctx)
-	if fegId == nil {
-		ctxMetadata, _ := metadata.FromIncomingContext(ctx)
-		errorStr := fmt.Sprintf(
-			"Failed to get Identity of calling Federated Gateway from CTX Metadata: %+v",
-			ctxMetadata,
-		)
-		glog.Error(errorStr)
-		return fmt.Errorf(errorStr)
+func getHwIDFromTeid(ctx context.Context, teid string) (string, error) {
+	gw := protos.GetClientGateway(ctx)
+	servedIds, err := getFegServedIds(gw.GetNetworkId())
+	if err != nil {
+		return "", err
 	}
-	if !fegId.Registered() {
-		return fmt.Errorf("federated gateway not registered")
+	for _, nid := range servedIds {
+		hwID, err := directoryd.GetHWIDForSgwCTeid(nid, teid)
+		if err == nil && len(hwID) != 0 {
+			glog.V(2).Infof("TEID to send is %s\n", teid)
+			return hwID, nil
+		}
```

Comment:
this is pretty heavy, it might be better in the future to instead add directoryd function to take a list of AGW network IDs and do it in one gRPC call. Maybe a separate PR.

Reply:
that will have to be changed here

http://github.com/magma/magma/blob/master/orc8r/cloud/go/services/directoryd/storage/storage_blobstore.go#L128-L128

Will require a change of protos to include a list of networks instead of string. 

Will raise another PR